### PR TITLE
DOC: fix broken link in inset-locator example

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Made a small error in #11026 with resolving a link in `examples/axes_grid1/inset_locator_demo.py`.  I think this fixes it...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/axes_grid1/inset_locator_demo.py [hunk 1]
```diff
@@ -6,11 +6,11 @@
 """
 
 ###############################################################################
-# The `.inset_locator`'s `~.axes_grid1.inset_axes` allows to easily place
-# insets in the corners of the axes by specifying a width and height and
-# optionally a location (loc) which accepts locations as codes, similar to
-# `~matplotlib.axes.Axes.legend`.
-# By default, the inset is offset by some points from the axes - this is
+# The `.inset_locator`'s `~mpl_toolkits.axes_grid1.inset_axes` allows
```
Comment (ImportanceOfBeingErnest) [inline examples/axes_grid1/inset_locator_demo.py]: If I'm not mistaken it should be 
`~mpl_toolkits.axes_grid1.inset_locator.inset_axes`, which may probably be shortened to `~inset_locator.inset_axes`. So I guess I was the one to actually make this mistake when I last edited that example.

**Comment by jklymak** [comment]
It worked before because `~.inset_axes` was unique in the namespace, now there is `Axes.inset_axes` (on master).

**Comment by ImportanceOfBeingErnest** [comment]
...a quite unfortunate decision to name it exactly like the existing one.

**Comment by jklymak** [comment]
The end goal is to deprecate axes_grid1, so I’m not concerned about the method having the same name.  Same thing happened w twinx.

**Pull request merged**